src/java.base/share/classes/java/util/zip/GZIPInputStream.java

```diff
@@ -34,8 +34,17 @@
 import java.util.Objects;
 
 /**
- * This class implements a stream filter for reading compressed data in
- * the GZIP file format.
+ * This class implements a stream filter for reading compressed data in the GZIP file format.
+ *
+ * <p>
+ * The GZIP compressed data format is self-delimiting, i.e., it includes an explicit GZIP trailer
+ * frame that marks the end of the compressed data. Therefore it's possible for the underlying
+ * input to contain additional data beyond the end of the compressed GZIP data. In particular,
+ * some GZIP compression tools function by partitioning the input, compressing each parttion
```

Comment:
I think we need to simplify the above  keeping the focus on that GZIPInputStream supports the reading of concatenated  Gzip streams.  After reading  a streams trailer(though some docs refer to it as a footer, but I think we go with trailer),  if an additional gzip header is found, the stream will be decompressed... etc.. 

If there is additional data/byes after the trailer that does not represent another gzip, header, it will indicate you have reached the end of the input stream

Reply:
Sounds good. How would you like to do that?

E.g. we could just remove the words "In particular, some GZIP compression tools function by partitioning the input, compressing each partition separately, and then concatenating the resulting compressed data streams. To support this kind of input".  We could also remove "In the latter case, the number of additional bytes that were read is unspecified." Or something else (what would you suggest?)

Thanks.

Reply:
> Sounds good. How would you like to do that?
> 
> E.g. we could just remove the words "In particular, some GZIP compression tools function by partitioning the input, compressing each partition separately, and then concatenating the resulting compressed data streams. To support this kind of input". We could also remove "In the latter case, the number of additional bytes that were read is unspecified." Or something else (what would you suggest?)
> 
> Thanks.

Yes, no reason to talk about gzip tools.  The focus should be that this method can read concatenated gzip streams when there is a gzip header immediately after the gzip trailer, otherwise additional bytes are treated as End of stream

Reply:
I'm reluctant to make this PR even longer lived. But perhaps this could be made even shorter by dropping the 'self-delimiting' term and its definition?

Could something like this work?

_In the GZIP file format, compressed data payloads are preceded by a header and followed by a trailer. If a new header immediately follows a trailer, this class continues to decode compressed data as a single, concatenated stream; otherwise, any additional bytes are ignored as if the end of stream is reached._